### PR TITLE
Added Azure.SQLMI.AADOnly

### LINKED ISSUE #2118: Check that Azure AD authentication is enforced for SQL Managed Instance
### Existing rule

_No response_

### Suggested rule

Azure SQL MI should be using Azure AD authentication. SQL legacy authentication mechanisms such as SQL logins are unable to provide sufficient protection for identities. Azure AD auth provides strong protection controls including conditional access, identity governance, and privileged identity management.

### Pillar

Security

### Additional context

- https://learn.microsoft.com/azure/azure-sql/database/authentication-azure-ad-only-authentication
- https://learn.microsoft.com/azure/azure-sql/database/authentication-aad-configure
- https://learn.microsoft.com/azure/architecture/framework/security/design-identity-authentication#use-modern-password-protection

### PR DESCRIPTION
**Pull request by BenjaminEngeset**
## PR Summary

Fixes #2118 

Added Azure.SQLMI.AADOnly rule.

## PR Checklist

- [x] PR has a meaningful title
- [x] Summarized changes
- [x] Change is not breaking
- [x] This PR is ready to merge and is not **Work in Progress**
- **Rule changes**
  - [x] Unit tests created/ updated
  - [x] Rule documentation created/ updated
  - [x] Link to a filed issue
  - [x] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section
- **Other code changes**
  - [ ] Unit tests created/ updated
  - [ ] Link to a filed issue
  - [ ] [Change log](https://github.com/Azure/PSRule.Rules.Azure/blob/main/docs/CHANGELOG-v1.md) has been updated with change under unreleased section

### CODE REVIEW

**Comment by BenjaminEngeset** [comment]
@BernieWhite Let me know.

File: docs/en/rules/Azure.SQLMI.AADOnly.md [hunk 1]
```diff
@@ -0,0 +1,147 @@
...
+
+## SYNOPSIS
+
+Ensure Azure AD-only authentication is enabled.
+
+## DESCRIPTION
+
+Azure AD authentication is a mechanism to connect to your SQL resource by using identities in Azure AD.
+
+SQL Managed Instance should be using Azure AD authentication. SQL legacy authentication mechanisms such as SQL logins are unable to provide sufficient protection for identities. Azure AD authentication provides strong protection controls including conditional access, identity governance, and privileged identity management.
```
Comment (BernieWhite) [inline docs/en/rules/Azure.SQLMI.AADOnly.md:19]:
```suggestion
Azure SQL Managed Instance supports authentication with SQL logins and Azure AD authentication.

By default, authentication with SQL logins is enabled.
SQL logins are unable to provide sufficient protection for identities.
Azure AD authentication provides strong protection controls including conditional access, identity governance, and privileged identity management.

Once you decide to use Azure AD authentication, you can disable authentication with SQL logins.
```

**Review by BernieWhite — CHANGES_REQUESTED** [review]
Great work @BenjaminEngeset! Just a few suggested updates. If you have any questions please let me know.

**Comment by BenjaminEngeset** [comment]
@BernieWhite I think we are all set, check the latest updates.

**Pull request merged**